### ISSUE TITLE
cannot import zone files greater than 8k bytes

### ISSUE THREAD

**Issue by johnjanicek** [WillyXJ/facileManager#363]
Using Admin->Tools->Import the zone file will load in the floating pop up window, BUT, after the popup footer for Import appears only a partial number of lines actually get imported which seems like it could be only the lines in the first 8K bytes of the imported zone file.  Would like to import a file with thousands of lines approximately 750K in size.  Previous instructions for .htaccess did not help:

 sw-engr-tulu:/srv/www/htdocs/facileManager # tail -3 .htaccess
\<IfModule mod_php5.c\>
php_value max_input_vars 256000
\</IfModule\>

Updating php config did not help either:
sw-engr-tulu:/etc/php5/apache2 # diff php.ini  .php.ini.jaj
393,394d392
< max_input_vars = 524288
< phpMaxDataSpace = 524288
797,798c795
< upload_max_filesize = 8M
< post_max_size = 8M
---
> upload_max_filesize = 2M

The values in 
Please help we will not be able to use this tool if we cannot import the existing config.
Our platform is running sles12sp2 + PHP 5.5.14 + apache2-2.4.23-16.3.x86_64.
And values in a local facileManager/phpinfo.php look correct.
I suspect that the function zoneImportWizard() begins processing lines but after the first 100 lines or so the remainder of the lines are ignored even though they can be seen in the pop up window.  Once the lines are imported the Records display does not contain all the lines and the remaining records do not appear in the database in the dns records:
MariaDB [facileManager]> describe fm_dns_records;
MariaDB [facileManager]> select count(*) from fm_dns_records;
+----------+
| count(*) |
+----------+
|       99 |
+----------+
1 row in set (0.00 sec)
Was trying to read in a forward zone file, but had the same experience reading in a reverse map.

**Comment by WillyXJ** [comment]
Hello,

Thank you for the report.  The Zone Import Wizard is undergoing big changes for the v3.0 release.  If you are up to testing the pre-release version, the wizard can handle a zone dump file (from `rndc dumpdb -zones`) which overcomes the PHP max_input_vars limitations.  You can download the pre-releases at http://www.facilemanager.com/download/?t=4.

If you need to stick with the current stable release, sadly the only workaround that I know of is to split up your zone and import in pieces.

**Comment by johnjanicek** [comment]
Excellent.  Sure, will give it a try and let you know how it works.  Thanks!

**Comment by johnjanicek** [comment]
Do you  know if there are any max parameter limits concerning the number of
lines (ie. buffer space) in the php pop up window?
I have verified that fm/fm-modules/fmDNS/classes/class_tools.php does in
fact read all the lines from a file, but, note that I had
to increase max_execution_time = 3600 and max_input_time = 3600 otherwise
the default time limit of 30 seconds would
prematurely stop processing of the input file.  I am seeing now that all
lines are read, but all the lines do not appear in the
pop up window that asks to 'Import' all the lines.  I have built in the
xdebug.org into my php environment and also loaded
the chrome browser extension to do real time debugging.  See var_dump lines
below.  I do not believe the pop up window
asking to Import the lines, does contain all the lines that were
processed.  I think lines are missing from the pop up window
and then when the Import is done, of course, lines are missing.  The import
only results in maybe 100-200 lines out of
what should have been 1000-2000 lines...

<snip>
                        if (empty($array['record_name']) &&
!empty($array['record_comment'])) continue;

                        $array['record_append'] =
(substr($array['record_value'], -1) == '.') ? 'no' : 'yes';

                        /** Set current_name to check for blanks on next
run */
                        $current_name = $array['record_name'];

                        $all_records[] = $array;

                        /** Automatically skip duplicates */
                        $checked = $this->checkDuplicates($array,
$domain_id, $all_records);
//                      unset($all_records);

var_dump( $count );
var_dump( $line );

                        if (!$get_dynamic_zone_data ||
($get_dynamic_zone_data && !$checked)) {
                                $rows .= $this->displayImportRow($array,
$record_action, $count, $checked);
                                $count++;
                        }
                }
<snip>

**Comment by WillyXJ** [comment]
I am not aware of anything that would limit the number of lines imported and still have a successful page load.  I just did an import of 20,000 records into a single zone (and the latest alpha release) and all records imported successfully (it took a while though).  I did have some issues at first as mod_security was blocking the large requests.  Perhaps you have something similar in your setup.

I made sure PHP wouldn't be limited so I used the following in my .htaccess (which is overkill):

```
php_value post_max_size 50M
php_value upload_max_filesize 50M
php_value max_input_vars 2000400
php_value max_execution_time 3600
php_value max_input_time 3600
```

If you get the zone file from a rndc dump and import it without selecting a zone, the import will be faster and not limited by any POST limitations.

**Comment by johnjanicek** [comment]
Removing this php module fixed the problem:

sw-engr-tulu:/etc/php5/apache2 # rpm -qa | grep -i suhosin
php5-suhosin-5.5.14-89.2.x86_64

With suhosin the $_POST variable was truncated prematurely equating to
about 100 lines of input...

My earlier comments were slightly wrong.  Both the main loop and pop up
window in class_tools.php did contains all
the input lines.  It was the post to zone-records-write.php where the
'$_POST' input string was truncated.
Adjusting the max_input_vars did  in the *.ini files and .htaccess did not
help.

I read the comment about suhosin on this link:

http://stackoverflow.com/questions/12684449/php-post-array-variables-are-truncated

I could see the post argument was truncated here via the var_dump debug
call:

./facileManager/fm-modules/fmDNS/pages/zone-records-write.php

<snip>
include(ABSPATH . 'fm-modules/fmDNS/classes/class_records.php');
if (empty($_POST)) header('Location: ' . $GLOBALS['RELPATH']);
var_dump ( $_POST );
/** Make sure we can handle all of the variables */
checkMaxInputVars();
extract($_POST);

So in summary, the two problems experienced were both config related issues:
1)  not enough time to import the file
     increase max_execution_time = 3600 and max_input_time = 3600
2)  bug created by php5-suhosin

I will continue testing the tool and let you know how it goes.  Have a good
weekend.

**Comment by WillyXJ** [comment]
Closing as the import tool in 3.0 and later and the mentioned system configuration changes addressed the issues reported here.
